File: widget_bridge/tags/version-1.1.2/demo_war/ajaxbridge.war/js/ContestChannel.js
/**
 * Copyright (C) 2008 TopCoder Inc., All Rights Reserved.
 *
 * Widget Bridge 1.0
 */

/**
 * The constructor takes JSON object and assign value to each internal fields.
 *
 * @param {JSONObject} json JSON object with will be converted to ContestChannel
 *
 * @consturctor
 * @namespace js.topcoder.widgets.bridge
 * @class ContestType
 *
 * This class is a client side mapping for ContestChannel entity in service side. It also encapsulates
 * JSON object coversion.
 * Thread Safety:
 * This class is mutable and not thread-safe. But JavaScript execution model is mostly single-threaded.
 * And there is no thread-synchronization features in JavaScript, so it is not an issue here. Anyway, the
 * external code should be careful if several threads are used to communicate with this class.
 *
 * @author TCSDEVELOPER
 * @version 1.0
 */
js.topcoder.widgets.bridge.ContestChannel = function(json){
    /**
     * This variable represents contest channel id. It can not be undefiend or null or less than -1 and its initial value
     *  is -1 which means it is not set yet.
     *
     * @private
     * @type number
     */
     var contestChannelID;

    /**
     * This variable represents contest type description. It can not be undefined or null.
     *
     * @private
     * @type string
     */
     var description;

    /**
     * Abbreviation for Helper class.
     */
    var Helper = js.topcoder.widgets.bridge.Helper;

    /**
     * Abbreviation for IllegalArgumentException class.
     */
    var IllegalArgumentException = js.topcoder.widgets.bridge.IllegalArgumentException;

    /**
     * A simple setter for contestChannelID internal field.
     *
     * @param {number} newContestChannelID the value of contest channel id.
     * @throws IllegalArgumentException if the argument is null or undefined value or less than 0.
     */
    this.setContestChannelID = function(newContestChannelID){
        Helper.checkGreaterOrEqual(newContestChannelID, "contest channel id", 0);
        contestChannelID = newContestChannelID;
    }

    /**
     * A simple getter for contestChannelId internal field.
     *
     * @return {number} contestChannelId internal field.
     */
    this.getContestChannelID = function(){
        return contestChannelID;
    }

    /**
     * A simple setter for description internal field.
     *
     * @param {string} newDescription the value of new description
     * @throws IllegalArgumentException if the argument is null or undefined value or not string type
     */
    this.setDescription = function(newDescription){
        Helper.checkDefined(newDescription, "description", "string");
        description = newDescription;
    }

    /**
     * A simple getter for description internal field.
     *
     * @return {string} description internal field.
     */
    this.getDescription = function(){
        return description;
    }

    /**
     * Convert the current instance into JSON string.
     *
     * @return a JSON string representing current object
     */
    this.toJSON = function(){
        var result = "{";
        result += "\"contestChannelID\":" + contestChannelID;
        result += ",\"description\":\"" + description+"\"";
		result += "}";
		return result;
    }

    //constructor code
    if (json != null) {
        this.setContestChannelId(json.contestChannelId);
        this.setDescription(json.description);
    }
}

File: contest_and_submission_entities/branches/spec_review2/src/java/main/com/topcoder/service/studio/contest/ContestChannel.java
/*
 * Copyright (C) 2009 TopCoder Inc., All Rights Reserved.
 */
package com.topcoder.service.studio.contest;

import java.io.Serializable;

/**
 * <p>
 * Represents the entity class for db table <i>contest_channel_lu</i>.
 * </p>
 *
 * <p>
 * Thread Safety: This entity is not thread safe since it is mutable.
 * </p>
 *
 * <p>
 * Version 1.1: removed name, parentChannelId and fileType fields.
 * </p>
 *
 * @author tushak, cyberjag
 * @version 1.1
 */
public class ContestChannel implements Serializable {
    /**
     * Generated serial version id.
     */
    private static final long serialVersionUID = 4713053485807111093L;

    /**
     * Represents the entity id.
     */
    private Long contestChannelId;

    /**
     * Represents the description of category.
     */
    private String description;

    /**
     * Default constructor.
     */
    public ContestChannel() {
        // empty
    }

    /**
     * Returns the contestChannelId.
     *
     * @return the contestChannelId.
     */
    public Long getContestChannelId() {
        return contestChannelId;
    }

    /**
     * Updates the contestChannelId with the specified value.
     *
     * @param contestChannelId
     *            the contestChannelId to set.
     */
    public void setContestChannelId(Long contestChannelId) {
        this.contestChannelId = contestChannelId;
    }

    /**
     * Returns the description.
     *
     * @return the description.
     */
    public String getDescription() {
        return description;
    }

    /**
     * Updates the description with the specified value.
     *
     * @param description
     *            the description to set.
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * Compares this object with the passed object for equality. Only the id will be compared.
     *
     * @param obj
     *            the {@code Object} to compare to this one
     * @return true if this object is equal to the other, {@code false} if not
     */
    @Override
    public boolean equals(Object obj) {
        if (obj instanceof ContestChannel) {
            return getContestChannelId() == ((ContestChannel) obj).getContestChannelId();
        }
        return false;
    }

    /**
     * Overrides {@code Object.hashCode()} to provide a hash code consistent with this class's
     * {@link #equals(Object)}} method.
     *
     * @return a hash code for this {@code ContestChannel}
     */
    @Override
    public int hashCode() {
        return Helper.calculateHash(contestChannelId);
    }
}
